[CI] IPSyntheticSourceNativeArrayIntegrationTests testSynthesizeArray failing

**Build Scans:**
- [elasticsearch-periodic #6636 / release-tests](https://gradle-enterprise.elastic.co/s/oiqrcvajhxoby)
- [elasticsearch-pull-request #58984 / part-1](https://gradle-enterprise.elastic.co/s/3i4fag3k2wyle)

**Reproduction Line:**
```
./gradlew ":server:test" --tests "org.elasticsearch.index.mapper.IPSyntheticSourceNativeArrayIntegrationTests.testSynthesizeArray" -Dtests.seed=405490CA8280498D -Dbuild.snapshot=false -Dtests.jvm.argline="-Dbuild.snapshot=false" -Dlicense.key=x-pack/license-tools/src/test/resources/public.key -Dtests.locale=en-AE -Dtests.timezone=Asia/Brunei -Druntime.java=23
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.index.mapper.IPSyntheticSourceNativeArrayIntegrationTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testSynthesizeArray),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: iterable with items ["_id", "_ignored_source"] in any order
     but: not matched: "_recovery_source"
```

**Issue Reasons:**
- [main] 2 failures in test testSynthesizeArray (5.1% fail rate in 39 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -337,15 +337,6 @@ tests:
 - class: org.elasticsearch.action.admin.indices.diskusage.IndexDiskUsageAnalyzerTests
   method: testCompletionField
   issue: https://github.com/elastic/elasticsearch/issues/123269
-- class: org.elasticsearch.index.mapper.IPSyntheticSourceNativeArrayIntegrationTests
-  method: testSynthesizeArray
-  issue: https://github.com/elastic/elasticsearch/issues/123417
-- class: org.elasticsearch.index.mapper.IPSyntheticSourceNativeArrayIntegrationTests
-  method: testSynthesizeArrayRandom
-  issue: https://github.com/elastic/elasticsearch/issues/123418
-- class: org.elasticsearch.index.mapper.IPSyntheticSourceNativeArrayIntegrationTests
-  method: testSynthesizeArrayIgnoreMalformed
-  issue: https://github.com/elastic/elasticsearch/issues/123419
 - class: org.elasticsearch.packaging.test.DockerTests
   method: test151MachineDependentHeapWithSizeOverride
   issue: https://github.com/elastic/elasticsearch/issues/123437

--- a/server/src/test/java/org/elasticsearch/index/mapper/NativeArrayIntegrationTestCase.java
+++ b/server/src/test/java/org/elasticsearch/index/mapper/NativeArrayIntegrationTestCase.java
@@ -141,7 +141,7 @@ public abstract class NativeArrayIntegrationTestCase extends ESSingleNodeTestCas
                 } else {
                     var copyExpectedStoredFields = new String[expectedStoredFields.length + 1];
                     System.arraycopy(expectedStoredFields, 0, copyExpectedStoredFields, 0, expectedStoredFields.length);
-                    copyExpectedStoredFields[copyExpectedStoredFields.length - 1] = "_ignored_source";
+                    copyExpectedStoredFields[copyExpectedStoredFields.length - 1] = "_recovery_source";
                     assertThat(storedFieldNames, containsInAnyOrder(copyExpectedStoredFields));
                 }
             }

